wit/preview2/redis.wit

```diff
@@ -0,0 +1,55 @@
+interface redis {
+  use fermyon:spin/redis-types.{payload, redis-parameter, redis-result}
+
+  /// Errors related to interacting with Redis
+  variant error {
+      /// An invalid address string
+      invalid-address,
+      /// There are too many open connections
+      too-many-connections,
+      /// A retrieved value was not of the correct type
+      type-error,
+      /// An I/O error occurred
+      io(string),
+  }
+  
+  resource connection {
+    /// Open a connection to the Redis instance at `address`.
+    open: static func(address: string) -> result<connection, error>
+
+    /// Publish a Redis message to the specified channel.
+    publish: func(channel: string, payload: payload) -> result<_, error>
+
+    /// Get the value of a key.
+    get: func(key: string) -> result<payload, error>
+
+    /// Set key to value.
+    ///
+    /// If key already holds a value, it is overwritten.
+    set: func(key: string, value: payload) -> result<_, error>
+
+    /// Increments the number stored at key by one.
+    ///
+    /// If the key does not exist, it is set to 0 before performing the operation.
+    /// An `error::type-error` is returned if the key contains a value of the wrong type
+    /// or contains a string that can not be represented as integer.
+    incr: func(key: string) -> result<s64, error>
+
+    /// Removes the specified keys.
+    ///
+    /// A key is ignored if it does not exist.
+    del: func(keys: list<string>) -> result<s64, error>
```

Comment:
Should this return a `u64`? Also maybe update the docs that it returns "the number of keys removed"?

Reply:
Good catch - this is copied from the first version. It seems that technically the returned number is a signed integer so the conservative thing to do would be to pass that along. However, we can be sure (right?!) that the number of keys has to be positive...

Reply:
Unless you start deleting anti-keys...

Reply:
I think it's fine to `u32`. Any concerns with that?